--- a/s2e/Makefile
+++ b/s2e/Makefile
@@ -247,14 +247,13 @@ $(Z3_BUILD_DIR):

 # Download SOCI
 $(SOCI_BUILD_DIR):
-	git clone $(SOCI_GIT_URL) $(SOCI_SRC_DIR)
-	cd $(SOCI_SRC_DIR) && git checkout $(SOCI_GIT_REV)
+	cp -r $(INJECTED_SOCI_SRC) $(SOCI_SRC_DIR)
 	mkdir -p $(S2E_BUILD)/$(SOCI_BUILD_DIR)

 # Download GTest
 $(GTEST_BUILD_DIR):
 	mkdir -p "$(GTEST_SRC_DIR)"
-	cd $(S2E_BUILD) && wget -O $(GTEST_SRC_DIR).tar.gz $(GTEST_URL) || rm -f "$@"
+	cd $(S2E_BUILD) && cp $(INJECTED_GTEST_SRC) $(GTEST_SRC_DIR).tar.gz
 	cd $(S2E_BUILD) && tar xzvf $(GTEST_SRC_DIR).tar.gz -C $(GTEST_SRC_DIR) --strip-components=1
 	mkdir -p "$@"

@@ -270,8 +269,7 @@ $(LIBDWARF_BUILD_DIR):
 	mkdir -p $(S2E_BUILD)/$(LIBDWARF_BUILD_DIR)

 $(RAPIDJSON_BUILD_DIR):
-	git clone $(RAPIDJSON_GIT_URL) $(RAPIDJSON_SRC_DIR)
-	cd $(RAPIDJSON_SRC_DIR) && git checkout $(RAPIDJSON_GIT_REV)
+	cp -r $(INJECTED_RAPIDJSON_SRC) $(RAPIDJSON_SRC_DIR)
 	mkdir -p $(S2E_BUILD)/$(RAPIDJSON_BUILD_DIR)

 $(PROTOBUF_BUILD_DIR):
@@ -478,6 +476,7 @@ stamps/protobuf-make: stamps/protobuf-configure
 #######

 stamps/lua-make: $(LUA_DIR)
+	$(SED) -i 's/gcc/clang/g' $(LUA_DIR)/src/Makefile
 	if [ "$(PLATFORM)" = "linux" ]; then \
 		$(SED) -i 's/-lreadline//g' $(LUA_DIR)/src/Makefile; \
 		$(MAKE) -C $^ linux CFLAGS="-DLUA_USE_LINUX -O2 -g -fPIC"; \
@@ -683,7 +682,7 @@ LIBS2E_CONFIGURE_FLAGS = --with-cc=$(CLANG_CC)
                          --with-z3-incdir=$(S2E_PREFIX)/include                     \
                          --with-z3-libdir=$(S2E_PREFIX)/lib                         \
                          --with-capstone-incdir=$(S2E_PREFIX)/include               \
-                         --with-capstone-libdir=$(S2E_PREFIX)/lib                   \
+                         --with-capstone-libdir=$(S2E_PREFIX)/lib64                 \
                          --with-libtcg-src=$(S2E_SRC)/libtcg                        \
                          --with-libcpu-src=$(S2E_SRC)/libcpu                        \
                          --with-libs2ecore-src=$(S2E_SRC)/libs2ecore                \
@@ -696,7 +695,7 @@ LIBS2E_DEBUG_FLAGS = --with-llvm=$(LLVM_BUILD)/llvm-debug
                      --with-fsigc++=$(S2E_BUILD)/libfsigc++-debug                   \
                      --with-libq=$(S2E_BUILD)/libq-debug                            \
                      --with-libcoroutine=$(S2E_BUILD)/libcoroutine-debug            \
-                     --with-cxxflags="$(CXXFLAGS_DEBUG)"                            \
+                     --with-cxxflags="$(INJECTED_LIBS2E_CXXFLAGS)"                  \
                      --enable-debug

 LIBS2E_RELEASE_FLAGS = --with-llvm=$(LLVM_BUILD)/llvm-release                       \
@@ -705,7 +704,7 @@ LIBS2E_RELEASE_FLAGS = --with-llvm=$(LLVM_BUILD)/llvm-release
                        --with-fsigc++=$(S2E_BUILD)/libfsigc++-release               \
                        --with-libq=$(S2E_BUILD)/libq-release                        \
                        --with-libcoroutine=$(S2E_BUILD)/libcoroutine-release        \
-                       --with-cxxflags="$(CXXFLAGS_RELEASE)"
+                       --with-cxxflags="$(INJECTED_LIBS2E_CXXFLAGS)"

 stamps/libs2e-debug-configure: $(call FIND_CONFIG_SOURCE,$(S2E_SRC)/libs2e)
 stamps/libs2e-debug-configure: stamps/lua-make stamps/libvmi-debug-install      \
@@ -758,7 +757,6 @@ stamps/libs2e-release-install: stamps/libs2e-release-make
 	install $(S2E_BUILD)/libs2e-release/i386-s2e_sp-softmmu/libs2e.so $(S2E_PREFIX)/share/libs2e/libs2e-i386-s2e_sp.so

 	install $(S2E_SRC)/libs2eplugins/src/s2e/Plugins/Support/KeyValueStore.py $(S2E_PREFIX)/bin/
-	cd $(S2E_SRC) && if [ -f ".git/config" ]; then git rev-parse HEAD > $(S2E_PREFIX)/share/libs2e/git-sha1; fi

 	touch $@

@@ -782,7 +780,6 @@ stamps/libs2e-debug-install: stamps/libs2e-debug-make
 	install $(S2E_BUILD)/libs2e-debug/i386-s2e_sp-softmmu/libs2e.so $(S2E_PREFIX)/share/libs2e/libs2e-i386-s2e_sp.so

 	install $(S2E_SRC)/libs2eplugins/src/s2e/Plugins/Support/KeyValueStore.py $(S2E_PREFIX)/bin/
-	cd $(S2E_SRC) && if [ -f ".git/config" ]; then git rev-parse HEAD > $(S2E_PREFIX)/share/libs2e/git-sha1; fi

 	touch $@

