--- a/setup.py
+++ b/setup.py
@@ -27,14 +27,9 @@ import pip
 from setuptools import setup, find_packages

 pyelftools_version = '0.24+s2e'
-pyelftools_url = f'git+https://github.com/S2E/pyelftools.git#egg=pyelftools-{pyelftools_version}'

 version = pip.__version__.split('.')
-if int(version[0]) < 19:
-    # For use together with --process-dependency-links
-    pyelftools = f'pyelftools=={pyelftools_version}'
-else:
-    pyelftools = 'pyelftools@%s' % pyelftools_url
+pyelftools = f'pyelftools=={pyelftools_version}'


 setup(
@@ -53,7 +48,6 @@ setup(

         # s2e-env requirements
         'distro',
-        'enum34',
         'jinja2',
         'pefile',
         'psutil',
@@ -85,9 +79,6 @@ setup(
         'mock',
     ],
     packages=find_packages(),
-    dependency_links=[
-        pyelftools_url,
-    ],
     include_package_data=True,
     package_data={
         's2e_env': [
